luxonis_ml/data/datasets/luxonis_dataset.py

```diff
@@ -503,9 +503,38 @@ def add(self, generator: DatasetIterator, batch_size: int = 1_000_000) -> Self:
 
         self.fs.put_file(tmp_file.name, "metadata/file_index.parquet")
         self._write_metadata()
-
+        self.warn_on_duplicates()
         return self
 
+    def warn_on_duplicates(self) -> None:
+        df = self._load_df_offline()
```

Comment:
Here we could make use of a lazy data frame for a better performance. Could you modify the `_load_df_offline` so it takes `lazy: bool = False` and returns a lazy frame if set to `True`? Check [`pl.scan_parquet`](https://docs.pola.rs/api/python/stable/reference/api/polars.scan_parquet.html).

Reply:
Made changes in the new commit